casr/src/bin/casr-cli.rs

```diff
@@ -824,12 +824,20 @@ fn print_summary(dir: &Path, unique_crash_line: bool) {
             .map(|res| res.unwrap().path())
             .any(|e| e.extension().is_some() && e.extension().unwrap() == "casrep")
     {
-        clusters.push((dir.to_path_buf(), 0));
+        if let Ok(canon_dir) = dir.canonicalize() {
+            clusters.push((canon_dir.to_path_buf(), 0));
+        } else {
+            clusters.push((dir.to_path_buf(), 0));
+        }
     }
 
     for (clpath, _) in clusters {
         let cluster = clpath.as_path();
-        let filename = cluster.file_name().unwrap().to_str().unwrap();
+        let filename = if let Some(cl_filename) = cluster.file_name() {
+            cl_filename.to_str().unwrap()
+        } else {
+            cluster.to_str().unwrap()
```

Comment:
why should be use full path here?

Reply:
If the filename is empty it means that it ends with '.' or is '/' and we couldn't canonicalize it. Then we have no other options besides using this path as is.

Reply:
> If the filename is empty it means that it ends with '.' or is '/' and we couldn't canonicalize it. Then we have no other options besides using this path as is.

Could you provide some use cases, when and why this situation happens?
It seems to me, that [this code](https://github.com/ispras/casr/blob/master/casr/src/bin/casr-cli.rs#L813) filter directories.

Reply:
The case is when casr-cli is called with an argument of directory name, and that directory contains raw casreps. For example, if casreps are located in root directory (a poked-out case, but still), or in cwd '.' and we couldn't canonicalize it for some reason. The bug was detected when calling casr-cli on cwd with raw reports.

Reply:
Can we add a comment in code stating this?

Reply:
Done